homeassistant/components/triggercmd/config_flow.py

```diff
@@ -0,0 +1,92 @@
+"""Config flow for TRIGGERcmd integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import jwt
+from triggercmd import TRIGGERcmdConnectionError, client, ha
+import voluptuous as vol
+
+from homeassistant import exceptions
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_TOKEN, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({(CONF_TOKEN): str})
+
+
+async def test_token(token: str) -> int:
+    """Test the auth token."""
+    r = await client.async_list(token)
+    return r.status_code
```

Comment:
Why do we now have so many extra functions?

Reply:
I moved this test to a separate function so I could easily mock it with pytest.  
I had to mock it because it does an HTTPS API call.